### PR TITLE
feat(cta): add SolidJS recipe

### PR DESCRIPTION
**Pull request by davedbase**
**What kind of change does this PR introduce? (check at least one)**

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change? (check one)**

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. fix: #xxx[,#xxx], where "xxx" is the issue number)
- [x]  A change file is added if any packages will require a version bump due to this PR per the instructions in the readme.

If adding a new feature, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: tooling/create-tauri-app/src/recipes/solid.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        when: !ci
+      }
+    ]
+  },
+  configUpdate: ({ cfg, packageManager }) => ({
+    ...cfg,
+    distDir: `../public`,
+    devPath: 'http://localhost:3000',
+    beforeDevCommand: `${
+      packageManager === 'yarn' ? 'npm run' : packageManager
```
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/solid.ts]:
```suggestion
      packageManager === 'npm' ? 'npm run' : packageManager
```

File: tooling/create-tauri-app/src/recipes/solid.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    } dev`,
+    beforeBuildCommand: `${
+      packageManager === 'yarn' ? 'npm run' : packageManager
```
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/solid.ts]:
```suggestion
      packageManager === 'npm' ? 'npm run' : packageManager
```

File: tooling/create-tauri-app/src/recipes/solid.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    )
+  },
+  postInit: async ({ cfg, packageManager }) => {
+    console.log(`
+    Your installation completed.
+    $ cd ${cfg.appName}
+    $ ${packageManager} install
+    $ ${packageManager === 'npm' ? 'npm run' : packageManager} tauri ${
+      packageManager === 'npm' ? '--' : ''
+    }dev
```
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/solid.ts]:
```suggestion
    $ ${packageManager === 'npm' ? 'npm run' : packageManager} tauri dev
```

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks for your efforts, there is a couple of things to change (I think you copied the recipe from `dev` branch instead of `next` which has these fixes already).

For tests, you can add it here https://github.com/tauri-apps/tauri/blob/next/tooling/create-tauri-app/test/spawn.test.mjs#L27 and here https://github.com/tauri-apps/tauri/blob/next/.github/workflows/test-cta.yml#L8

**Comment by amrbashir** [comment]
I almost forgot, the PR is missing a changefile and please fill out the PR template instead of removing it.

**Comment by davedbase** [comment]
My apologies @amrbashir I threw this up too quickly then rushed away to another task without reading the procedure properly. I've changed the description with the proper PR template, changefile (I looked at Svelte's changefile) and applied the changes you requested.

File: .github/workflows/test-cta.yml [hunk 4]
```diff
@@ -5,30 +5,30 @@
...
   RUST_BACKTRACE: 1
-  TAURI_RECIPE: 'vanillajs,cra,vite,ngcli'
+  TAURI_RECIPE: 'vanillajs,cra,vite,ngcli,solid'
 
 on:
   workflow_dispatch:
     inputs:
       platform:
-        default: "ubuntu"
+        default: 'ubuntu'
```
Comment (amrbashir) [inline .github/workflows/test-cta.yml:14]: Can you revert these changes ? probably your IDE auto format. 

I won't point out all similar formatting changes but the file has a few more of this.

File: tooling/create-tauri-app/src/recipes/solid.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      packageManager === 'npm' ? 'npm run' : packageManager
+    } dev`,
+    beforeBuildCommand: `${
+      packageManager === 'npm' ? 'npm run' : packageManager
+    } build`
+  }),
+  preInit: async ({ cwd, cfg, answers }) => {
+    await shell(
+      'npx',
+      ['degit', `solidjs/templates/${answers?.template}`, `${cfg.appName}`],
```
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/solid.ts]:
```suggestion
      ['degit', `solidjs/templates/${answers?.template ?? 'js'}`, cfg.appName],
```
This will ensure that there is a template when `ci` mode is on, otherwise the resulting string would be `'solidjs/templates/undefined'`

**Review by amrbashir — CHANGES_REQUESTED** [review]
No need to apologize.

Just some more small changes and this PR is ready to go.

File: tooling/create-tauri-app/src/recipes/solid.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+import { join } from 'path'
+import { shell } from '../shell'
+import { Recipe } from '../types/recipe'
+
+const solid: Recipe = {
+  descriptiveName: {
+    name: 'Solid (https://github.com/solidjs/solid)',
```
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/solid.ts]: This should link to the templates Repo.

File: tooling/create-tauri-app/src/recipes/solid.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  shortName: 'solid',
+  extraNpmDevDependencies: [],
+  extraNpmDependencies: [],
+  extraQuestions: ({ ci }) => {
+    return [
+      {
+        type: 'list',
+        name: 'template',
+        message: 'Which Solid template would you like to use? (Read more at https://github.com/solidjs/templates)',
```
Comment (amrbashir) [inline tooling/create-tauri-app/src/recipes/solid.ts]: With the requested change above, we should remove the link from here.

**Review by amrbashir — CHANGES_REQUESTED** [review]
Sorry I forgot about these.

File: .github/workflows/test-cta.yml (hunk 4 above)
Comment (davedbase) [reply .github/workflows/test-cta.yml:14]: Yeah VSCode has a plugin for pretty YAML. Sigh. This is now reverted.

File: tooling/create-tauri-app/src/recipes/solid.ts (hunk 7 above)
Comment (davedbase) [reply tooling/create-tauri-app/src/recipes/solid.ts]: Absolutely. Yeah I see Svelte is doing the same thing. Done!

**Comment by davedbase** [comment]
@amrbashir my apologies this PR could have gone a lot smoother. I hope things are looking much better now!

File: .github/workflows/test-cta.yml [hunk 8]
```diff
@@ -14,21 +14,21 @@ on:
         default: "ubuntu"
```
Comment (amrbashir) [inline .github/workflows/test-cta.yml]:
```suggestion
        default: 'ubuntu'
```

**Review by amrbashir — COMMENTED** [review]
yeah the PR looks good to me now but since we are in a feature-freeze phase, it might be a while before it gets merged.

**Review by amrbashir — APPROVED** [review]

**Comment by davedbase** [comment]
Yup no worries! Looking forward to when its merged.

**Comment by davedbase** [comment]
@amrbashir should I be concerned that it's not passing the eslint test? I adjusted the types to mitigate it but it's still stumbling....

**Comment by amrbashir** [comment]
> @amrbashir should I be concerned that it's not passing the eslint test? I adjusted the types to mitigate it but it's still stumbling....

That is actually a formatting issue, should be fixed by running `yarn format`.

**Comment by davedbase** [comment]
@amrbashir the test runner didn't list any formatting errors. When I ran prettier it didn't change much except removed end of line. *shrugs* lol
